### PR TITLE
[incubator-kie-issues#1187] Split EvalHelper

### LINKED ISSUE #1187: DMN: Split EvalHelper to better focused classes
Current EvalHelper does not have a clearly defined scope.
Split it to specialized classes

### PR DESCRIPTION
**Pull request by gitgabrio**
Fixes https://github.com/apache/incubator-kie-issues/issues/1187

This PR split the EvalHelper class to better focused helpers

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Review by baldimir — APPROVED** [review]

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FEELDialect.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
```
Comment (yesamer) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FEELDialect.java:11]: @gitgabrio Those <p> tags are unnecessary

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FEELDialect.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * under the License.
+ */
+package org.kie.dmn.feel.lang;
+
+import java.util.Arrays;
+
+public enum FEELDialect {
+
+    FEEL(""),
+    BFEEL("https://www.omg.org/spec/DMN/20240513/B-FEEL/");
```
Comment (yesamer) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FEELDialect.java:26]: @gitgabrio Could make sense to add a comment here stating that is an experimental (or WIP) feature?

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/FEELBuilder.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
```
Comment (yesamer) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/FEELBuilder.java:11]: those <p> tags are unnecessary

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/StringEvalHelper.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StringEvalHelper {
+    public static final Logger LOG = LoggerFactory.getLogger( StringEvalHelper.class );
+
+    public static String normalizeVariableName(String name) {
+        // private static final Pattern SPACES_PATTERN = Pattern.compile( "[\\s\u00A0]+" );
+        // return SPACES_PATTERN.matcher( name.trim() ).replaceAll( " " );
```
Comment (yesamer) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/StringEvalHelper.java:29]: Are those commented lines still necessary?

**Review by yesamer — APPROVED** [review]
Nice refactoring!

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/StringEvalHelper.java (hunk 4 above)
Comment (gitgabrio) [reply kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/StringEvalHelper.java:29]: See the comment soon below - maybe it is worth keep it for reference, at least for the moment being

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/FEELBuilder.java (hunk 3 above)
Comment (gitgabrio) [reply kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/FEELBuilder.java:11]: Fixed in new PR

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FEELDialect.java (hunk 2 above)
Comment (gitgabrio) [reply kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FEELDialect.java:26]: @baldimir ^^ ?

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FEELDialect.java (hunk 1 above)
Comment (gitgabrio) [reply kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/FEELDialect.java:11]: Fixed in new PR

**Pull request merged**